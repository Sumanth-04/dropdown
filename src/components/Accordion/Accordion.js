import React, { Component } from 'react';
import './index.scss';
import PropTypes from 'prop-types';


class Accordion extends Component {
	static propTypes = {
		onClick: PropTypes.func,
		children: PropTypes.instanceOf(Array).isRequired
	}


	constructor(props) {
		super(props);


		this.state = {
			activeTab: this.props.children[0].props.label,
		};
	}


	onClickTabItem = (tab) => {
		this.setState({ activeTab: tab });
	}


	render() {
		const { children } = this.props;
		const { activeTab } = this.state;


		return (
			<div className={``}>
			{
					children.map((child) => {
						const { label, action, headingImage, headingBody , accordionColor} = child.props;
						
						return (
							<div>
								{/* accordion header */}
								<div className={`accordion`} style={{background:accordionColor}} onClick={() => { this.onClickTabItem(label) }}>
									<div className="rd-accordionHeader">
										<div>{label}</div>
										{headingBody ? <div>{headingBody}</div> : null}
									</div>
									<div>{headingImage ? <img className={`rd-accordionImage`} src={`${headingImage}`} alt="accordion" /> : null}</div>
									
									{/* accordion body */}
									<div className='rd-accordionBody'>
									{
										child.props.label == activeTab ? child.props.children : undefined
									}
								</div>
								</div>
							</div>
						);
					})
				}
			</div>

		)
	}
}



export default Accordion;




// Â®
	// children.map((child) => {
	// 	const { label, action, headingImage, headingBody } = child.props;
	// 	return (
	// 		<div>
	// 			{/* accordion header */}
	// 			<div className="accordion" onClick={() => { this.onClickTabItem(label) }}>
	// 				<div className="rd-accordionHeader">
	// 					<div>{label}</div>
	// 					{headingBody ? <div>{headingBody}</div> : null}
	// 				</div>
	// 				<div>{headingImage ? <img className={`rd-accordionImage`} src={`${headingImage}`} alt="accordion" /> : null}</div>
	// 			</div>
	// 			{/* accordion body */}
	// 			{
	// 				child.props.label == activeTab ? child.props.children : undefined
	// 			}
	// 		</div>
	// 	);
	// })
