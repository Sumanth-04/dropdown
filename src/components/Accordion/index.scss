/* Style the buttons that are used to open and close the accordion panel */
/* .accordion {
	background-color: #eee;
	color: #444;
	cursor: pointer;
	width: 100%;
	padding: 22px 40px;
	text-align: left;
	border: none;
	outline: none;
	transition: 0.4s;
  } */
  
  
  /* Add a background color to the button if it is clicked on (add the .active class with JS), and when you move the mouse over it (hover) */
  /* .active, .accordion:hover {
	background-color: #ccc;
  } */
  
  /* Style the accordion panel. Note: hidden by default */
  /* .panel {
	padding: 0 18px;
	background-color: white;
	overflow: hidden;
  }
	.rd-accordionHeader {
		display: flex;
		justify-content: space-between;
	} */

	
	@import '../variables';
	/*Accordion*/
	
	.accordion {
		background:$accordion-background;
		color: $accordion-color;
		cursor: pointer;
		width: 100%;
		padding: 22px 40px;
		margin-top: 1em;
		text-align: left;
		border: none;
		outline: none;
		transition: 0.4s;
	}
	
	/* Add a background color to the button if it is clicked on (add the .active class with JS), and when you move the mouse over it (hover) */
	.active,
	.accordion:hover {
		background: $accordion-background-hover;
	}
	
	/* Style the accordion panel. Note: hidden by default */
	.panel {
		padding: 10px 18px;
		background-color: $panel-background;
		overflow: hidden;
	}
	.rd-accordionHeader {
		display: flex;
		justify-content: space-between;
		color: $accordionHeader-color;
		font-size: $accordionHeader-fontsize;
	}
	
	.rd-accordionBody {
		font-family: Chivo;
	}
	


	


	.container {
		margin: 1% 3%;
		font-family: Chivo;
	  }
	  .container-heading {
		font-family: ChivoLight;
		font-size: 22px;
	  }
	  .specSection{
		margin-top: 3em;
		background: #FDFDFD;
	   
	  }
	  .featureSection{
		padding: 1em 1em 2em 2em;
		font-size: 18px;
	  }
	  
	  /*specSectionFeatures*/
	  
	  /*Main Container*/
	  .specSectionFeatures{
		font-family: ChivoLight;
		font-size: 12px;
		height: inherit;
	  }
	  /*Each div*/
	  .text-center{
		float: right;
		padding-right: 4%;
	  }
	  /*Img inside Each div*/
	  .rd-opacity-6{
		color: black;
		opacity: 0.6;
	  }
	  /*Text inside each div*/
	  .specText{
		text-align: center;
		color: black;
	  }
	  .caret{
		float: right;
		padding-right: 1%;
	  }
	  .caretText{
		font-size: 20px;
		color: grey;
	  }