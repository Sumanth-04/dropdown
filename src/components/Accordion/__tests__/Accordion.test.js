import React from "react";
import renderer from 'react-test-renderer';
import Accordion from "../Accordion";
//import {isTSAnyKeyword} from '@babel/types';

it("Accordion should have a single child and options in that child (Use div tag)", () => {
  const tree = renderer
    .create(
      <Accordion>
        <div key='1' label="HEllo">
          HEllo Body
        </div>
        <div key='2' label="Hi">
          Hi Body
        </div>
        <div key='3' label="Good">
          GOod Body
        </div>
      </Accordion>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();

  /*const div = document.createElement("div");
  ReactDOM.render(
        
        , div
    );*/
});