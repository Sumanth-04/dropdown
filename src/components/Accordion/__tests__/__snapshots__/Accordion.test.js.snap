// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion should have a single child and options in that child (Use div tag) 1`] = `
<div
  className=""
>
  <div>
    <div
      className="accordion"
      onClick={[Function]}
      style={
        Object {
          "background": undefined,
        }
      }
    >
      <div
        className="rd-accordionHeader"
      >
        <div>
          HEllo
        </div>
      </div>
      <div />
      <div
        className="rd-accordionBody"
      >
        HEllo Body
      </div>
    </div>
  </div>
  <div>
    <div
      className="accordion"
      onClick={[Function]}
      style={
        Object {
          "background": undefined,
        }
      }
    >
      <div
        className="rd-accordionHeader"
      >
        <div>
          Hi
        </div>
      </div>
      <div />
      <div
        className="rd-accordionBody"
      />
    </div>
  </div>
  <div>
    <div
      className="accordion"
      onClick={[Function]}
      style={
        Object {
          "background": undefined,
        }
      }
    >
      <div
        className="rd-accordionHeader"
      >
        <div>
          Good
        </div>
      </div>
      <div />
      <div
        className="rd-accordionBody"
      />
    </div>
  </div>
</div>
`;

exports[`Accordion should have a single child and options in that child (Use div tag) 2`] = `
<div
  className=""
>
  <div>
    <div
      className="accordion"
      onClick={[Function]}
      style={
        Object {
          "background": undefined,
        }
      }
    >
      <div
        className="rd-accordionHeader"
      >
        <div>
          HEllo
        </div>
      </div>
      <div />
      <div
        className="rd-accordionBody"
      >
        Lorem Body
      </div>
    </div>
  </div>
  <div>
    <div
      className="accordion"
      onClick={[Function]}
      style={
        Object {
          "background": undefined,
        }
      }
    >
      <div
        className="rd-accordionHeader"
      >
        <div>
          HEllo
        </div>
      </div>
      <div />
      <div
        className="rd-accordionBody"
      >
        Lorem Epseuim
      </div>
    </div>
  </div>
</div>
`;
